File: src/ros2_bridge/mappers/nav2_msgs/route_edge.rs
//! Typed mapper for `nav2_msgs/msg/RouteEdge`.

use crate::ros2_bridge::mapper::TypedTopicMapper;

pub(crate) fn route_edge_to_bus(msg: ros_env::nav2_msgs::msg::RouteEdge) -> crate::nav2_msgs::msg::v1::RouteEdge {
    crate::nav2_msgs::msg::v1::RouteEdge {
        edgeid: msg.edgeid.into(),
        start: msg.start.into(),
        end: msg.end.into(),
    }
}

pub(crate) fn route_edge_to_ros(bus: crate::nav2_msgs::msg::v1::RouteEdge) -> ros_env::nav2_msgs::msg::RouteEdge {
    ros_env::nav2_msgs::msg::RouteEdge {
        edgeid: bus.edgeid as _,
        start: bus.start as _,
        end: bus.end as _,
    }
}

#[derive(Clone, Copy, Debug, Default)]
pub struct Nav2MsgsRouteEdgeMapper;

impl TypedTopicMapper for Nav2MsgsRouteEdgeMapper {
    type Ros = ros_env::nav2_msgs::msg::RouteEdge;
    type Bus = crate::nav2_msgs::msg::v1::RouteEdge;

    fn ros_to_bus(&self, msg: Self::Ros) -> crate::errors::Result<Self::Bus> {
        Ok(route_edge_to_bus(msg))
    }

    fn bus_to_ros(&self, msg: Self::Bus) -> crate::errors::Result<Self::Ros> {
        Ok(route_edge_to_ros(msg))
    }
}

File: bindings/python/robot_bus/ros2_bridge/mappers/nav2_msgs/route_edge.py
"""Generated mapper for `nav2_msgs/msg/RouteEdge`."""

from __future__ import annotations

from robot_bus.ros2_bridge.mappers import _convert

_BusMsg = None


def _bus_cls():
    global _BusMsg
    if _BusMsg is None:
        from robot_bus.nav2_msgs.msg.v1 import RouteEdge as BusMsg

        _BusMsg = BusMsg
    return _BusMsg


def route_edge_to_bus(msg):
    BusMsg = _bus_cls()
    bus = BusMsg()
    bus.edgeid = msg.edgeid
    bus.start = msg.start
    bus.end = msg.end
    return bus


def route_edge_to_ros(bus):
    from nav2_msgs.msg import RouteEdge as RosMsg

    out = RosMsg()
    out.edgeid = bus.edgeid
    out.start = bus.start
    out.end = bus.end
    return out


class Nav2MsgsRouteEdgeMapper:
    _ros_type = None

    def ros_msg_type(self):
        cls = type(self)
        if cls._ros_type is None:
            from nav2_msgs.msg import RouteEdge as RosMsg

            cls._ros_type = RosMsg
        return cls._ros_type

    def ros_to_bus(self, msg) -> bytes:
        return route_edge_to_bus(msg).SerializeToString()

    def bus_to_ros(self, payload: bytes):
        BusMsg = _bus_cls()
        bus = BusMsg()
        bus.ParseFromString(payload)
        return route_edge_to_ros(bus)
